test/TEST-55-OOMD/test.sh

```diff
@@ -5,18 +5,19 @@ set -e
 TEST_DESCRIPTION="systemd-oomd Memory Pressure Test"
 IMAGE_NAME="oomd"
 
+# Need to set up swap
+TEST_NO_NSPAWN=1
+
 # shellcheck source=test/test-functions
 . "${TEST_BASE_DIR:?}/test-functions"
 
 test_append_files() {
-    # Create a swap device
+    # Create a swap file
     (
-        mkswap "${LOOPDEV:?}p2"
-        image_install swapon swapoff
+        image_install mkswap swapon swapoff stress
 
-        cat >>"${initdir:?}/etc/fstab" <<EOF
-UUID=$(blkid -o value -s UUID "${LOOPDEV}p2")    none    swap    defaults 0 0
-EOF
+        dd if=/dev/zero of="${initdir:?}/swapfile" bs=1M count=48
```

Comment:
why not just call `truncate`?

Reply:
just following the manpage suggestion

Reply:
Also, `mkswap` will complain if you use a sparse file:

```
$ truncate -s 50M ~/tmp/foo
$ mkswap ~/tmp/foo 
mkswap: /home/fsumsal/tmp/foo: insecure permissions 0644, fix with: chmod 0600 /home/fsumsal/tmp/foo

mkswap: /home/fsumsal/tmp/foo contains holes or other unsupported extents.
        This swap file can be rejected by kernel on swap activation!
        Use --verbose for more details.

Setting up swapspace version 1, size = 50 MiB (52424704 bytes)
no label, UUID=e7e51d40-a2d5-4215-a56b-2bee4c1da7e9

```